website/client/src/components/auth/register.vue

```diff
@@ -0,0 +1,343 @@
+        </router-link>
+      </div>
+    </div>
+  </form>
+</template>
+
+<style lang="scss" scoped>
+  @import '~@/assets/scss/colors.scss';
+
+  .register-form {
+    margin: 0 auto;
+    width: 40em;
+    position: relative;
+    z-index: 1;
+
+    ::-webkit-input-placeholder { /* Chrome/Opera/Safari */
+      color: $purple-400;
+    }
+    ::-moz-placeholder { /* Firefox 19+ */
+      color: $purple-400;
+    }
+    :-ms-input-placeholder { /* IE 10+ */
+      color: $purple-400;
+    }
+    :-moz-placeholder { /* Firefox 18- */
+      color: $purple-400;
+    }
+    ::placeholder { //  Standard browsers
+      color: $purple-400;
+    }
+
+    label {
+      color: $white;
+      font-weight: bold;
+    }
+
+    input {
+      margin-bottom: 2em;
+      border-radius: 2px;
+      background-color: #432874;
+      border-color: transparent;
+      height: 50px;
+      color: $white;
+    }
+
+    .input-with-error.input-invalid {
+      margin-bottom: 0.5em;
+    }
+
+    #confirmPasswordInput + .input-error {
+      margin-bottom: 2em;
+    }
+
+    .form-text {
+      font-size: 14px;
+      color: $white;
+    }
+
+    .toggle-links {
+      margin-top: 1em;
+    }
+
+    .toggle-link {
+      color: $white !important;
+    }
+
+    .input-error {
+      color: #fff;
+      font-size: 90%;
+      width: 100%;
+    }
+  }
+
+  .register-form-plansPage {
+    .form-group {
+      text-align: left;
+      font-weight: bold;
+    }
+
+    small.form-text {
+      text-align: center;
+    }
+  }
+
+  .register-form-homePage {
+    padding-bottom: 1em;
+
+    input {
+      margin-bottom: 1em;
+      border-radius: 2px;
+      background-color: #432874;
+      border-color: #432874;
+      color: $purple-400;
+      border: solid 2px transparent;
+      transition-timing-function: ease;
+      transition: border .5s, color .5s;
+    }
+
+    .sign-up {
+      padding-top: 11px;
+      padding-bottom: 11px;
+    }
+
+    h1, h2, h3, h4, h5, h6, button, .strike > span, input {
+      font-family: 'Varela Round', sans-serif;
+      font-weight: normal;
+    }
+  }
+</style>
+
+<script>
+import debounce from 'lodash/debounce';
+import isEmail from 'validator/lib/isEmail';
+import { MINIMUM_PASSWORD_LENGTH } from '@/../../common/script/constants';
+
+export default {
+  data () {
+    const data = {
+      username: '',
+      email: '',
+      password: '',
+      passwordConfirm: '',
+      usernameIssues: [],
+    };
+
+    return data;
+  },
+  computed: {
+    isHomePage () {
+      if (this.$route.path.startsWith('/static/home')) {
+        return true;
+      }
+      return false;
+    },
+    isRegisterPage () {
+      if (this.$route.path.startsWith('/register')) {
+        return true;
+      }
+      return false;
+    },
+    isPlansPage () {
+      if (this.$route.path.startsWith('/static/plans')) {
+        return true;
+      }
+      return false;
+    },
+    usernameValid () {
```

Comment:
Here too I would recommend not checking the path of the current page. It is knowledge this abstraction is not supposed to have (And makes future changes in the other pages likely to cause bugs in this component without the person changing that page realising it). A quick win would be to replace these computed values with injected properties just like you did for social-auth.

And here also I'd recommend going for descriptive properties instead of consumption-driven properties. A large part of the home page variation seems to be a "showLabels" variation. However remember that you can also let parents apply styling to divs in child components using [deep selectors](https://vue-loader.vuejs.org/guide/scoped-css.html#deep-selectors). You could for example do something like:

```css
// pseudo-css since I don't know what the actual id of surrounding element is
#register-form >>> label {
  display: none;
}
```
This could also be a solution for the "in what page am I being displayed"-specific css.

Depending on which CSS pre-processor Habitica uses you may need to use one of the other deep selectors (`/deep/` or `::v-deep`) but the link should have the info you need.

Reply:
`::v-deep` should be used